### PR TITLE
feat(#2663): support for playwright-core usage

### LINKED ISSUE #2663: Support for playwright-core
#### Proposal
Need support for `playwright-core` for projects where browser binaries are managed separately and don't need the full weight of installing `playwright`. 

https://github.com/microsoft/playwright/tree/master/packages/playwright-core

### PR DESCRIPTION
**Pull request by andremoah**
## Motivation/Description of the PR
- Resolves #2663 .

Added a simple utility that allows the require fallback of npm packages. If the first is not found it will try the next one in the chain.
I have been using it to include playwright-core instead of playwright because I don't want to download the browser binaries on yarn/npm install. This approach can be used for other helpers as well if you guys think it's worth it.

For example, it would be possible as well for puppeteer without adding 

```
npm config set puppeteer_skip_chromium_download true
```

as explained in https://github.com/codeceptjs/CodeceptJS/issues/2076
 

Applicable helpers:

- [ ] WebDriver
- [ ] Puppeteer
- [ ] Nightmare
- [ ] REST
- [ ] FileHelper
- [ ] Appium
- [ ] Protractor
- [ ] TestCafe
- [x] Playwright

Applicable plugins:

- [ ] allure
- [ ] autoDelay
- [ ] autoLogin
- [ ] customLocator
- [ ] pauseOnFail
- [ ] coverage
- [ ] retryFailedStep
- [ ] screenshotOnFail
- [ ] selenoid
- [ ] stepByStepReport
- [ ] stepTimeout
- [ ] wdio
- [ ] subtitles

## Type of change

- [ ] :fire: Breaking changes
- [x] :rocket: New functionality
- [ ] :bug: Bug fix
- [ ] :clipboard: Documentation changes/updates
- [ ] :hotsprings: Hot fix
- [ ] :hammer: Markdown files fix - not related to source code
- [ ] :nail_care: Polish code

## Checklist:

- [x] Tests have been added
- [ ] Documentation has been added (Run `npm run docs`)
- [x] Lint checking (Run `npm run lint`)
- [x] Local tests are passed (Run `npm test`)

### CODE REVIEW

**Review by Arhell — APPROVED** [review]

**Comment by DavertMik** [comment]
Thanks, this is a good idea. Probably the installation of playwright-core should be documented as well...
Does this solution is also applicable for puppeteer & puppeteer code?

**Comment by andremoah** [comment]
> Thanks, this is a good idea. Probably the installation of playwright-core should be documented as well... Does this solution is also applicable for puppeteer & puppeteer code?

Yes totally! the logic is exactly the same, I can do some extended tests to see if it behaves the same way as Playwright. About the documentation, do you mean in https://codecept.io/ or in the code? I am open to contributing to that, just please let me know  how can i help best.

**Comment by DavertMik** [comment]
I think you can update documentation inside the code of Playwright helper. In the very beginning of a file.
And yes, please make the same fix (with the same docs update) for Puppeteer

**Comment by andremoah** [comment]
> I think you can update documentation inside the code of Playwright helper. In the very beginning of a file. And yes, please make the same fix (with the same docs update) for Puppeteer

Updated docs and added also puppeteer-core support.

**Comment by DavertMik** [comment]
Thank you!

File: lib/helper/Playwright.js [hunk 1]
```diff
@@ -49,12 +50,16 @@ const { createValueEngine, createDisabledEngine } = require('./extras/Playwright
  *
  * This helper works with a browser out of the box with no additional tools required to install.
  *
- * Requires `playwright` package version ^1 to be installed:
+ * Requires `playwright` or `playwright-core` package version ^1 to be installed:
  *
  * ```
  * npm i playwright@^1 --save
+ *
```
Comment (DavertMik) [inline lib/helper/Playwright.js:57]:
```suggestion
 * ```
 * or
 * ```
```

File: lib/helper/Playwright.js [hunk 2]
```diff
@@ -49,12 +50,16 @@ const { createValueEngine, createDisabledEngine } = require('./extras/Playwright
...
+ * npm i playwright-core@^1 --save
  * ```
  *
+ * Using the core version of playwright package, will prevent download of browser binaries and allows you to connect to an existing browser installation or for connecting to a remote one.
```
Comment (DavertMik) [inline lib/helper/Playwright.js:61]:
```suggestion
 * Using playwright-core package, will prevent the download of browser binaries and allow connecting to an existing browser installation or for connecting to a remote one.
```

File: lib/helper/Puppeteer.js [hunk 3]
```diff
@@ -43,7 +44,13 @@ const consoleLogStore = new Console();
  * Browser control is executed via DevTools Protocol (instead of Selenium).
  * This helper works with a browser out of the box with no additional tools required to install.
  *
- * Requires `puppeteer` package to be installed.
+ * Requires `puppeteer` or `puppeteer-core` package to be installed.
+ * ```
+ * npm i puppeteer --save
+ *
```
Comment (DavertMik) [inline lib/helper/Puppeteer.js:50]:
```suggestion
 * ```
 * or
 * ```
```

File: lib/helper/Puppeteer.js [hunk 4]
```diff
@@ -43,7 +44,13 @@ const consoleLogStore = new Console();
...
+ * npm i puppeteer-core --save
+ * ```
+ * Using the core version of puppeteer package, will prevent download of browser binaries and allows you to connect to an existing browser installation or for connecting to a remote one.
```
Comment (DavertMik) [inline lib/helper/Puppeteer.js:53]:
```suggestion
 * Using `puppeteer-core` package, will prevent the download of browser binaries and allow connecting to an existing browser installation or for connecting to a remote one.
```

**Review by DavertMik — APPROVED** [review]

**Pull request merged**